packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -141,11 +151,17 @@ class CodeSizeController {
   void loadDiffTreeFromJsonFiles(
     DevToolsJsonFile oldFile,
     DevToolsJsonFile newFile,
+    void Function(String error) onError,
   ) {
-    if (oldFile == null || newFile == null) return;
+    if (oldFile == null || newFile == null) {
+      return;
+    }
 
-    changeOldDiffSnapshotFile(oldFile);
-    changeNewDiffSnapshotFile(newFile);
+    if (oldFile.isApkFile != newFile.isApkFile) {
+      onError(
+        'Failed to load snapshot: Given snapshots to compare are of different types.',
```

Comment:
"Failed to load diff: OLD and NEW files are different types."

Reply:
Done